perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/ConsentsManagerMethod.java

```diff
@@ -1,15 +1,92 @@
 package cz.metacentrum.perun.rpc.methods;
 
+import cz.metacentrum.perun.core.api.Consent;
+import cz.metacentrum.perun.core.api.ConsentStatus;
 import cz.metacentrum.perun.core.api.exceptions.PerunException;
 import cz.metacentrum.perun.rpc.ApiCaller;
 import cz.metacentrum.perun.rpc.ManagerMethod;
 import cz.metacentrum.perun.rpc.deserializer.Deserializer;
 
+import java.util.List;
+
 public enum ConsentsManagerMethod implements ManagerMethod {
-	;
 
-	@Override
-	public Object call(ApiCaller ac, Deserializer parms) throws PerunException {
-		return null;
+	/*#
+	 * Returns all existing Consents
+	 *
+	 * @return List<Consent> list of Consents
+	 */
+	getAllConsents {
+		@Override
+		public List<Consent> call (ApiCaller ac, Deserializer parms) throws PerunException {
+			return ac.getConsentsManager().getAllConsents(ac.getSession());
+		}
+	},
+
+	/*#
+	 * Returns Consent with the corresponding id
+	 *
+	 * @param id int Consent <code>id</code>
+	 * @throw ConsentNotExistsException when Consent specified by <code>id</code> doesn't exist
+	 * @return List<Consent> list of Consents
+	 */
+	getConsentById {
+		@Override
+		public Consent call (ApiCaller ac, Deserializer parms) throws PerunException {
+			return ac.getConsentsManager().getConsentById(ac.getSession(), parms.readInt("id"));
+		}
+	},
+
+	/*#
+	 * Returns all consents for a User
+	 *
+	 * @param id int User <code>id</code>
+	 * @throw UserNotExistsException when User specified by <code>id</code> doesn't exist
+	 * @return List<Consent> Consents of the User
+	 */
+	/*#
+	 * Returns all consents with a specific status for a User.
+	 *
+	 * @param id int User <code>id</code>
+	 * @param status String UNSIGNED | GRANTED | REVOKED
+	 * @throw UserNotExistsException when User specified by <code>id</code> doesn't exist
+	 * @return List<Consent> Consents of the User
+	 */
+	getConsentsForUser {
+		@Override
+		public List<Consent> call (ApiCaller ac, Deserializer parms) throws PerunException {
+			if (parms.contains("status")) {
+				return ac.getConsentsManager().getConsentsForUser(ac.getSession(), parms.readInt("id"), ConsentStatus.valueOf(parms.readString("status")));
```

Comment:
beware how you name the input parameters in openapi, you should use existing reference to userId, which uses name `user` - you will need to rename the `id` to `user` here then..